app/src/main/java/woowacourse/shopping/view/product/catalog/ProductCatalogViewModelFactory.kt

```
@@ -0,0 +1,25 @@
+    private val recentProductRepository: RecentProductRepository,
+) : ViewModelProvider.Factory {
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        if (modelClass.isAssignableFrom(ProductCatalogViewModel::class.java)) {
+            return ProductCatalogViewModel(
+                productRepository,
+                cartProductRepository,
+                recentProductRepository,
+            ) as T
+        }
+        throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)